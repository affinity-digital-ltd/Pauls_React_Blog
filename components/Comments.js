import { Component } from 'react'

class Comments extends Component {
  render () {
    return <div className='c-comment'>
      <div className='c-comment__author'>
        <img className='c-comment__img' src='../static/images/ruby.jpg' alt='A Ruby logo' />
        <h4 className='c-comment__author-name'>Author Name</h4>
        <time className='c-comment__created-date'>Created on: Add Date</time>
      </div>
      <p className='c-comment__body'>Body text</p>
    </div>
  }
}

export default Comments
