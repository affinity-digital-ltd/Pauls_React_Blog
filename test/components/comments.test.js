import Comments from '../../components/Comments'

describe('<Comments />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Comments />
    )

    expect(component).toMatchSnapshot()
  })
})