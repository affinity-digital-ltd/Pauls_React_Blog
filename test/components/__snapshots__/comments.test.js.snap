// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-comment__author"
        >
          <img
            alt="A Ruby logo"
            className="c-comment__img"
            src="../static/images/ruby.jpg"
          />
          <h4
            className="c-comment__author-name"
          >
            Author Name
          </h4>
          <time
            className="c-comment__created-date"
          >
            Created on: Add Date
          </time>
        </div>,
        <p
          className="c-comment__body"
        >
          Body text
        </p>,
      ],
      "className": "c-comment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="A Ruby logo"
              className="c-comment__img"
              src="../static/images/ruby.jpg"
            />,
            <h4
              className="c-comment__author-name"
            >
              Author Name
            </h4>,
            <time
              className="c-comment__created-date"
            >
              Created on: Add Date
            </time>,
          ],
          "className": "c-comment__author",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "A Ruby logo",
              "className": "c-comment__img",
              "src": "../static/images/ruby.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Author Name",
              "className": "c-comment__author-name",
            },
            "ref": null,
            "rendered": "Author Name",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Created on: Add Date",
              "className": "c-comment__created-date",
            },
            "ref": null,
            "rendered": "Created on: Add Date",
            "type": "time",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Body text",
          "className": "c-comment__body",
        },
        "ref": null,
        "rendered": "Body text",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-comment__author"
          >
            <img
              alt="A Ruby logo"
              className="c-comment__img"
              src="../static/images/ruby.jpg"
            />
            <h4
              className="c-comment__author-name"
            >
              Author Name
            </h4>
            <time
              className="c-comment__created-date"
            >
              Created on: Add Date
            </time>
          </div>,
          <p
            className="c-comment__body"
          >
            Body text
          </p>,
        ],
        "className": "c-comment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="A Ruby logo"
                className="c-comment__img"
                src="../static/images/ruby.jpg"
              />,
              <h4
                className="c-comment__author-name"
              >
                Author Name
              </h4>,
              <time
                className="c-comment__created-date"
              >
                Created on: Add Date
              </time>,
            ],
            "className": "c-comment__author",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "A Ruby logo",
                "className": "c-comment__img",
                "src": "../static/images/ruby.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Author Name",
                "className": "c-comment__author-name",
              },
              "ref": null,
              "rendered": "Author Name",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Created on: Add Date",
                "className": "c-comment__created-date",
              },
              "ref": null,
              "rendered": "Created on: Add Date",
              "type": "time",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Body text",
            "className": "c-comment__body",
          },
          "ref": null,
          "rendered": "Body text",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
